nixos/modules/services/monitoring/thanos.nix

```diff
@@ -0,0 +1,770 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.thanos;
+
+  nullOpt = type : description : mkOption {
+    type = types.nullOr type;
+    default = null;
+    inherit description;
+  };
+
+  optionToArgs = opt : v   : optional (v != null)  ''--${opt}="${toString v}"'';
+  flagToArgs   = opt : v   : optional v            ''--${opt}'';
+  listToArgs   = opt : vs  : map               (v: ''--${opt}="${v}"'') vs;
+  attrsToArgs  = opt : kvs : mapAttrsToList (k: v: ''--${opt}=${k}=\"${v}\"'') kvs;
+
+  mkParamDef = type : default : description : mkParam type (description + ''
```

Comment:
I'm not really sure about the space before the column. Although this does look nice, I have not seen many examples of it.

Reply:
I see this is indeed not a common style in nixpkgs. I'll remove the space before all colons.